src/Cmfd.cpp

```diff
@@ -368,21 +345,21 @@ void Cmfd::computeDs(int moc_iteration) {
 
           /* Get diffusivity and flux for Mesh cell */
           d = _materials[cell]->getDifCoef()[e];
-          flux = _old_flux[cell*_num_cmfd_groups+e];
+          flux = _old_flux->getValue(cell, e);
           cell_next = getCellNext(cell, surface);
 
           /* Set halfspace sense of the Surface */
-          if (surface == 0 || surface == 1)
+          if (surface == SURFACE_X_MIN || surface == SURFACE_Y_MIN)
             sense = -1.0;
           else
             sense = 1.0;
 
           /* Set the length of this Surface and the perpendicular Surface */
-          if (surface == 0 || surface== 2) {
+          if (surface == SURFACE_X_MIN || surface== SURFACE_X_MAX) {
```

Comment:
Cmfd requires that the Cmfd lattice be a regular grid. In other words, the cell width in a particular direction must be the same for all cells, but the cell width is not necessarily the same for all directions. For most CMFD cases (e.g. c5g7-cmfd.py) `length` and `length_perpen` will be the same irregardless of the value of `surface`. However, it the Cmfd lattice in `c5g7-cmfd.py` was changed from 51x51 to 51x3, `length` and `length_perpen` would be depend on whether `surface` is an x surface or a y surface.

The challenge with changing the surface enums from being unique for each surface to having two enums to identify the surface direction (x or y) and side (min or max) is that corners would require two directions and two sides. Also, this would increase the storage requirement for each segment to at least two enums. I only see one conditional block with an `X_MIN' or 'X_MAX' condition and would side on keeping the constants as is.